sdk/log/processor.go

```diff
@@ -44,9 +49,14 @@ type Processor interface {
 	// indeterminate state, but may return false if valid reasons in particular
 	// circumstances exist (e.g. performance, correctness).
 	//
-	// Before modifying a Record, the implementation must use Record.Clone
+	// The SDK invokes the processors sequentially in the same order as
+	// they were registered using [WithProcessor] until any processor returns true.
+	// Implementations may synchronously modify the record so that the changes
+	// are visible in the next registered processor.
+	// Notice that [Record] is not concurrent safe. Therefore, asynchronous
+	// processing may cause race conditions. Use [Record.Clone]
 	// to create a copy that shares no state with the original.
-	Enabled(ctx context.Context, record Record) bool
+	Enabled(ctx context.Context, record *Record) bool
```

Comment:
Why is Enabled passed by reference?  We don't expect the record to be modified during the Enabled call, right?

Reply:
I found this odd as well. The more I think about it, the more I don't think it should be passed as a pointer similar to @dashpole.

- The method does not mutate a record
- The API does not accept a pointer
- We should not encourage users to mutate the record during the call
- The record pointer is discarded after the call to `Enabled` meaning the reference will be garbage collected
- If we return `false` there is a good chance the user will not emit the record, meaning we will be creating a heap allocations that are avoidable.

Reply:
Gotcha. Will change back to value tomorrow.

Reply:
I think we can also make some changes in Record to make it more immutable but I will leave it to a separate PR

Reply:
PTAL https://github.com/open-telemetry/opentelemetry-go/pull/5636/commits/dfa72a60aabcdc6f8141f0bd901caa9f228fe1f3

(working on updating the benchmark results)

Reply:
@dmathieu, can you also double-check after the changes?

Reply:
Having an immutable record in `Enabled()` definitely makes more sense.